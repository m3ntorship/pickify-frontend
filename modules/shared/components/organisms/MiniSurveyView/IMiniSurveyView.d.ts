import type { IPostFeed } from '../../../types/postFeed/IPostFeed';

declare namespace IMiniSurveyView {
  export interface IProps {
    post: IPostFeed.IPost;
    deletePostHandler: (postId: string) => void;
    addOneVote: (optionId: string, grouId: string) => void;
    sharePostHandler: (postId: string) => void;
  }
}

export { IMiniSurveyView };
