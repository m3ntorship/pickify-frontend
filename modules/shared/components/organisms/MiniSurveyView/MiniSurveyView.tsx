import React from 'react';
import type { FC, ReactElement } from 'react';
import Image from 'next/image';
import type { IPostFeed } from '../../../types/postFeed/IPostFeed';
import MiniSurveyViewOptions from '../../molecules/MiniSurveyViewOptions/MiniSurveyViewOptions';
import PostViewHeader from '../../molecules/PostViewHeader/PostViewHeader';
import PostViewFooter from '../../molecules/postFooter/PostFooter';
import type { IMiniSurveyView } from './IMiniSurveyView';
import { getVotesResults } from '../../../logic/votesLogic/votesLogic';

const MiniSurveyView: FC<IMiniSurveyView.IProps> = ({
  post,
  addOneVote,
  deletePostHandler,
}): ReactElement => {
  let votedOptions: IPostFeed.IOptions[] = [];

  post.options_groups.groups.map((group) => {
    const votes = group.options.map((option) => option);
    votedOptions = [...votedOptions, ...votes];
    return group;
  });

  const showTotalVotes = (): boolean => {
    const totalVotes = votedOptions.filter(
      (option) => option.vote_count !== undefined,
    );
    if (totalVotes.length > 0) {
      return true;
    }
    return false;
  };

  const { totalVotes } = getVotesResults(votedOptions);
  const { user } = post;
  return (
    <div className="bg-white p-m shadow-soft rounded-md space-y-4" id={post.id}>
      <PostViewHeader
        postId={post.id}
        userId={user ? user.id : ''}
        date={new Date(post.created_at)}
        name={user ? user.name : undefined}
        profilePic={user ? user.profile_pic : undefined}
        isHidden={post.is_hidden}
        deletePostHandler={deletePostHandler}
      />
      <div>
        <h3 className="font-normal text-md">{post.caption}</h3>
      </div>
      {post.media.length !== 0 && (
        <>
          {post.media.map((image) => (
            <div key={image.url} className="relative">
              <div className="absolute w-full h-full rounded-md overflow-hidden">
                <Image
                  src={image.url}
                  layout="responsive"
                  className="filter blur-sm"
                  objectFit="cover"
                  width={600}
                  height={600}
                />
              </div>
              <Image
                src={image.url}
                layout="responsive"
                className="rounded-md"
                objectFit="contain"
                width={600}
                height={600}
              />
            </div>
          ))}
        </>
      )}
      <MiniSurveyViewOptions
        optionsGroups={post.options_groups}
        addOneVote={addOneVote}
      />
      <PostViewFooter
        numberOfVotes={totalVotes}
        showResult={showTotalVotes()}
      />
    </div>
  );
};
export default MiniSurveyView;
