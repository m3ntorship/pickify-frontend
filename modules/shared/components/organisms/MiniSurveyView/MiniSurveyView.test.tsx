import React from 'react';
import * as renderer from 'react-test-renderer';
import type { IPostFeed } from '../../../types/postFeed/IPostFeed';
import MiniSurveyView from './MiniSurveyView';

jest.mock('../../../logic/formatDate/getNow');

describe('ITextPollView', () => {
  const mockedPost: IPostFeed.IPost = {
    caption: 'nalyzing Delaware Frozen',
    id: '03644270-7171-4147-b5a1-4233ff547f7ddda',
    created_at: '2020-12-31T00:00:00.000Z',
    user: {
      name: 'Ahmed Ayoub',
      id: '465456',
      profile_pic:
        'https://i.pinimg.com/736x/10/61/c8/1061c85ea0dfd3a6655b4b3b6e71ade9.jpg',
    },
    type: 'text poll',
    is_hidden: false,
    media: [{ url: 'http://placeimg.com/640/480/nature?t=1622888620897' }],
    options_groups: {
      groups: [
        {
          id: '03644270-7171-4147-b5a1-4233ff547f7a',
          name: 'Group Name',
          options: [
            {
              id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
              body: 'ali tramsy',
              media: [{ url: 'https://placeimg.com/640/480/any' }],
            },
            {
              id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
              body: ' ahmed essam',
              media: [{ url: 'https://placeimg.com/640/480/any' }],
            },
            {
              id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
              body: ' omar gamal',
              media: [{ url: 'https://placeimg.com/640/480/any' }],
            },
          ],
        },
      ],
    },
  };
  it('should render ITextPollView Compnent with the mocked data', () => {
    const tree = renderer.create(<MiniSurveyView post={mockedPost} />);
    expect(tree).toMatchSnapshot();
  });
});
