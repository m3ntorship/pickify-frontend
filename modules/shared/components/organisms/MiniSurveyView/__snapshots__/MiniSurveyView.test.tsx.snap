// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ITextPollView should render ITextPollView Compnent with the mocked data 1`] = `
<div
  className="bg-white p-m shadow-soft rounded-md space-y-4"
  id="03644270-7171-4147-b5a1-4233ff547f7ddda"
>
  <div>
    <div
      className="flex justify-between items-start pb-s"
    >
      <div
        className="outer-wrapper"
      >
        <div
          className="wrapper medium"
        >
          <svg
            className="absolute w-full h-full"
            data-file-name="SvgNotFilled"
            height={40}
            onClick={[Function]}
            width={40}
          />
        </div>
        <div
          className="user-wrapper"
        >
          <span
            className="name"
            data-testid="name"
          >
            ayoub
          </span>
          <span
            className="date"
            title="31/12/2020 00:00:00"
          >
            a day ago
          </span>
        </div>
      </div>
      <svg
        className="fill-grey cursor-pointer w-6 h-6"
        data-file-name="SvgVerticalThreeDots"
        data-testid="03644270-7171-4147-b5a1-4233ff547f7ddda"
        onClick={[Function]}
      />
    </div>
    <div
      className="bg-grey-shd6 h-px"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div>
    <h3
      className="font-normal text-md"
    >
      nalyzing Delaware Frozen
    </h3>
  </div>
  <div
    className="space-y-4"
  >
    <div
      className="space-y-2"
    >
      <div
        className="text-dark-grey text-sm"
      >
        Group Name
      </div>
      <div
        data-testid="option"
      >
        <button
          className="btnBody"
          data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
          disabled={false}
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
          onClick={[Function]}
          type="button"
        >
          <span
            className="flex-container"
          >
            <span
              className="letter"
            >
              a
            </span>
            <p
              className="option"
            >
               
              ali tramsy
               
            </p>
            <span />
          </span>
        </button>
      </div>
      <div
        data-testid="option"
      >
        <button
          className="btnBody"
          data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa"
          disabled={false}
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa"
          onClick={[Function]}
          type="button"
        >
          <span
            className="flex-container"
          >
            <span
              className="letter"
            >
              b
            </span>
            <p
              className="option"
            >
               
               ahmed essam
               
            </p>
            <span />
          </span>
        </button>
      </div>
      <div
        data-testid="option"
      >
        <button
          className="btnBody"
          data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
          disabled={false}
          id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
          onClick={[Function]}
          type="button"
        >
          <span
            className="flex-container"
          >
            <span
              className="letter"
            >
              c
            </span>
            <p
              className="option"
            >
               
               omar gamal
               
            </p>
            <span />
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <span>
       Vote to uncover the total number of voters 
    </span>
    <button
      type="button"
    >
      <svg
        data-file-name="SvgShare"
      />
    </button>
  </div>
</div>
`;
